File: android/app/src/main/java/com/pizzadelivery/android/data/model/MenuItem.kt
package com.pizzadelivery.android.data.model

import com.google.gson.annotations.SerializedName

data class MenuItem(
    val id: String = "",
    val storeId: String = "",
    val name: String = "",
    val description: String = "",
    val price: Double = 0.0,
    val category: String = "",
    val image: String = "",
    val customizations: List<Customization> = emptyList(),
    @SerializedName("isAvailable")
    val isAvailable: Boolean = true,
    val rating: Double = 0.0,
    val calories: Int? = null
)

data class Customization(
    val id: String = "",
    val name: String = "",
    val type: CustomizationType = CustomizationType.SINGLE,
    val required: Boolean = false,
    val options: List<CustomizationOption> = emptyList()
)

enum class CustomizationType {
    @SerializedName("single") SINGLE,
    @SerializedName("multiple") MULTIPLE
}

data class CustomizationOption(
    val id: String = "",
    val name: String = "",
    val price: Double = 0.0
)

data class MenuCategory(
    val id: String = "",
    val name: String = "",
    val description: String = "",
    val image: String = "",
    val itemCount: Int = 0
)

File: ios/PizzaDelivery/Sources/Models/MenuItem.swift
import Foundation

struct MenuItem: Codable, Identifiable {
    let id: String
    var storeId: String
    var name: String
    var description: String
    var price: Double
    var category: String
    var image: String?
    var customizations: [Customization]?
    var isAvailable: Bool
    
    enum CodingKeys: String, CodingKey {
        case id, storeId, name, description, price, category, image, customizations, isAvailable
    }
}

struct Customization: Codable, Identifiable {
    var id: String { name }
    var name: String
    var options: [CustomizationOption]
    var required: Bool
    var maxSelections: Int?
}

struct CustomizationOption: Codable, Identifiable, Hashable {
    var id: String { name }
    var name: String
    var price: Double
    
    func hash(into hasher: inout Hasher) {
        hasher.combine(name)
    }
    
    static func == (lhs: CustomizationOption, rhs: CustomizationOption) -> Bool {
        lhs.name == rhs.name && lhs.price == rhs.price
    }
}

struct MenuCategory: Codable, Identifiable {
    var id: String { name }
    var name: String
    var itemCount: Int?
}
